### DISCUSSION TITLE
maximum number of files and folders which can be compared?

### DISCUSSION THREAD

**Discussion by libove** [WinMerge/winmerge#1958] [General]
Due to a OneDrive for Business/ Sharepoint corruption, I may need to download two copies of folder trees, each to contain over 100,000 files distributed across a few thousand folders, holding about a terabyte total of data. This is almost entirely photo/video/sound media data, so comparisons need to be binary byte-for-byte until either end-of-file is matched or any difference at all is found.
How many folders and files can WinMerge handle? Could it compare two folder trees, each containing a terabyte or so of data, spread over more than 100,000 files throughout thousands of subfolders?
Alternatively, it might be practical for me to script a few hundred command line calls to WinMerge, each call to compare one subfolder tree; in these cases, each subfolder tree pair to be compared should contain no more than thousands of files in dozens of folders.
thanks for your thoughts,

**Comment by sdottaka** [comment]
WinMerge should be able to handle around 100,000 files. WinMerge does not impose specific constraints, so with sufficient memory, it should be able to compare up to the limit of 32-bit integers.

When performing comparisons, to increase the speed of the process, it is recommended to set the comparison method to "Binary Contents," as follows.

![image](https://github.com/WinMerge/winmerge/assets/98126/b786bc42-a839-43e9-8a45-02d2f20e4b21)
